features/polyPinion/src/context/questionnaire-list-context.tsx

```diff
@@ -89,18 +94,19 @@ export const QuestionnaireListProvider: React.FC = ({ children }) => {
 
     const getNewActiveQuestionnairesMetadata = async () => {
         const resultValue = await downloadActiveQuestionnairesMetadata();
-        const allActiveQuestionnairesMetadata = JSON.parse(resultValue);
-        if (
-            allActiveQuestionnairesMetadata === null ||
-            allActiveQuestionnairesMetadata.length === 0
-        ) {
+        // TO-FIX: define a proper type for this JSON string object
+        // we should know format of questionnairesMetadata
+        const allActiveQuestionnairesMetadataJSON = JSON.parse(resultValue);
+        if (allActiveQuestionnairesMetadataJSON === null) {
             return [];
         }
 
-        const questionnairesIndex = questionnaireList.map((questionnaire) => questionnaire.id);
+        const questionnairesIndexes = questionnaireList.map((questionnaire) => questionnaire.id);
```

Comment:
Nit: I think the idea with naming this was that this is "an index of all the questionnaires", not "the indexes of all questionnaires", since I believe questionnaires don't have indexes - they have IDs.

Reply:
also, plural is "indices" :sweat_smile:

Reply:
I looked it up, seems both is actually correct and widely used :D